src/app.py

```diff
@@ -0,0 +1,72 @@
+import datetime
+
+from email_validator import EmailNotValidError, validate_email
+from flask import Flask, render_template, request
+from flask_sqlalchemy import SQLAlchemy
+
+app = Flask(__name__)
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///ttmn.db"
+
+db = SQLAlchemy()
+db.init_app(app)
+
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String, unique=True, nullable=False)
+    time = db.Column(db.Time(timezone=True), default=datetime.time(9))
+
+
+# Create tables if they don't exist
+with app.app_context():
+    db.create_all()
+
+
+@app.route("/")
+def home():
+    return render_template("index.html")
+
+
+@app.route("/register", methods=("GET", "POST"))
+def register():
+    error = False
+    if request.method == "POST":
+        # Get and validate the email
+        email = request.form["email"]
+        try:
+            validate_email(email)
+        except EmailNotValidError:
+            error = "Invalid email"
+
+        # Check if the email is already used
+        used = User.query.filter_by(email=email).first()
+        if used:
+            error = "Email already used"
+
+        # Get and validate the time
+        time = request.form["timezone"]
+        try:
+            time = datetime.datetime.strptime(time, "%H").time()
+        except ValueError:
+            error = "Invalid time"
+
+        if not error:
+            # Create the user
+            user = User(
+                email=email,
+                time=time,
+            )
+
+            # Add the user to the database
+            db.session.add(user)
+            db.session.commit()
+
+            return render_template("news.html")
+
+    return render_template("signup.html", error=error)
+
+
+@app.route("/news")
+def news():
+    posts = save_posts()
+    return render_template("news.html", posts=posts)
```

Comment:
why not have this in main.py and have the app.run stuff at the bottom like in the previous main.py

Reply:
It looks like running the code as a normal Python file is not recommended:

> The reason for this is that due to how the reload mechanism works there are some bizarre side-effects (like executing certain code twice, sometimes crashing without message or dying when a syntax or import error happens).

I'm not sure that this is related to that, but, for example, based on where you run `app.py` from, the database gets generated in a different directory.